src/layers/icmpv6.rs

```diff
@@ -176,28 +222,87 @@ impl Layer for ICMPv6 {
             }
 
             ICMPV6_ROUTER_ADVERTISEMENT => {
-                let mut flags = RouterAdvFlags::default();
                 let cur_hop_limit = u8::from_be(bytes[4]);
-                flags.managed_address_flag = ((bytes[5] >> 7) & 0x01) == 0x01;
-                flags.other_config = ((bytes[5] >> 6) & 0x01) == 0x01;
+                let flags = RouterAdvFlags {
+                    managed_address_flag: ((bytes[5] >> 7) & 0x01) == 0x01,
+                    other_config: ((bytes[5] >> 6) & 0x01) == 0x01,
+                };
 
                 let router_lifetime: u16 = (bytes[6] as u16) << 8 | (bytes[7] as u16);
                 let reachable_time: u32 = u32::from_be_bytes(bytes[8..12].try_into().unwrap());
                 let retrans_timer: u32 = u32::from_be_bytes(bytes[12..16].try_into().unwrap());
-                decoded = 16;
+
+                let option_type: u8 = u8::from_be(bytes[16]);
+                let length = u8::from_be(bytes[17]);
+                decoded = 17 + ((length << 3) as usize) - 2;
+                let src_link_layer = Icmpv6LinkLayerAddress {
+                    option_type,
+                    length,
+                    link_layer_address: bytes[18..decoded + 1].try_into().unwrap(),
+                };
+                decoded += 1;
+                let mtu = Icmpv6OptionsMTU {
```

Comment:
This should be one Variant of the `ICMPv6Option`. This will have following variants (eg. defined in the same RFC section 4.6). So this should look something like the following - 

enum IcmpV6Ooption { 
   LinkLayerAddress(LinkLayerAddress), // As per section 4.6.1
   PrefixInfo(PrefixInfo) // As per section 4.6.2
   RedirectHeader(RedirectHeader), // As per section 4.6.3 
   Mtu(Mtu), // As per section 4.6 .4
   Unsupported(Vec<u8>) // All options we do not parse yet (we can even have that as a structure with type, length identified and data as unsupported unparsed data). 
} 

And then your `IcmpV6RouterAdvertisement` - would look something like - 
```rust 
IcmpV6RouterAdvertisement {
  curr_hop_limit:..., 
  flags:..., 

  options: Vec<IcmpV6Option> // possibly empty 
}
```

Reply:
@gabhijit  sir, I have made the above mentioned changes. Please review it.
Thank you.